userspace/libsinsp/container.h

```diff
@@ -136,6 +175,7 @@ class sinsp_container_manager
 
 	sinsp* m_inspector;
 	std::unordered_map<std::string, std::shared_ptr<const sinsp_container_info>> m_containers;
+	std::unordered_map<std::string, std::unordered_map<sinsp_container_type, sinsp_container_lookup_state>> m_lookups;
```

Comment:
I think this would be simpler if you made a new LookupKey class (which was string and sinsp_container_type) and then this is just `std::unordered_map<LookupKey, lookup_state> m_lookups`